File: scripts/extract_cert/extract_cert.js
/**
 * Apenas extrai e exibe os campos do certificado (sem chamar o WS).
 */
import fs from "node:fs";
import path from "node:path";
import { fileURLToPath } from "node:url";
import dotenv from "dotenv";
import { extractFromPfx } from "../../lib/cert_extract.js";

const __dirname = path.dirname(fileURLToPath(import.meta.url));
const ROOT = path.resolve(__dirname, "../..");

dotenv.config({ path: path.join(ROOT, ".env") });

const certPath = process.env.CERT_PATH;
const certPassword = process.env.CERT_PASSWORD ?? "";

if (!certPath) {
  console.error("Defina CERT_PATH no arquivo .env");
  process.exit(1);
}

const fields = extractFromPfx(certPath, certPassword, {
  publickeyFormat: process.env.PUBLICKEY_FORMAT || "base64_der",
  validuntilFormat: process.env.VALIDUNTIL_FORMAT || "iso",
});

const preview = { ...fields };
preview.PUBLICKEY = `${preview.PUBLICKEY.slice(0, 48)}... (${preview.PUBLICKEY.length} chars)`;
delete preview._has_private_key;

console.log(JSON.stringify(preview, null, 2));

const outPath = path.join(ROOT, "cert-fields.json");
const full = { ...fields };
delete full._has_private_key;
fs.writeFileSync(outPath, JSON.stringify(full, null, 2), "utf-8");
console.log(`\nCampos completos salvos em: ${outPath}`);

File: scripts/extract_cert/extract_cert.py
#!/usr/bin/env python3
"""Apenas extrai e exibe os campos do certificado (sem chamar o WS)."""

from __future__ import annotations

import json
import os
import sys
from pathlib import Path

from dotenv import load_dotenv

ROOT = Path(__file__).resolve().parents[2]
sys.path.insert(0, str(ROOT))

from lib.cert_extract import extract_from_pfx  # noqa: E402


def main() -> int:
    load_dotenv(ROOT / ".env")

    cert_path = os.getenv("CERT_PATH")
    cert_password = os.getenv("CERT_PASSWORD", "")
    if not cert_path:
        print("Defina CERT_PATH no arquivo .env", file=sys.stderr)
        return 1

    fields = extract_from_pfx(
        cert_path,
        cert_password,
        publickey_format=os.getenv("PUBLICKEY_FORMAT", "base64_der"),
        validuntil_format=os.getenv("VALIDUNTIL_FORMAT", "iso"),
    )

    preview = {k: v for k, v in fields.items() if not k.startswith("_")}
    pk = preview["PUBLICKEY"]
    preview["PUBLICKEY"] = f"{pk[:48]}... ({len(pk)} chars)"
    print(json.dumps(preview, ensure_ascii=False, indent=2))

    out = ROOT / "cert-fields.json"
    full = {k: v for k, v in fields.items() if not k.startswith("_")}
    out.write_text(json.dumps(full, ensure_ascii=False, indent=2), encoding="utf-8")
    print(f"\nCampos completos salvos em: {out}")
    return 0


if __name__ == "__main__":
    raise SystemExit(main())
